shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm

```diff
@@ -436,8 +437,20 @@ - (nullable id)accessibilityElementAtIndex:(NSInteger)index {
     return _semanticsObject;
   }
   SemanticsObject* child = [_semanticsObject children][index - 1];
-  if ([child hasChildren])
-    return [child accessibilityContainer];
+    
+    // This 'if' block handles adding accessibility support for the embedded platform view.
+    // We first check if the child is a semantic node for a platform view.
+    // If so, we add the platform view as accessibilityElements of the child.
+    shell::FlutterPlatformViewsController *flutterPlatformViewsController = _bridge.get()->flutter_platform_views_controller();
+    if (child.node.platformViewId > -1 && flutterPlatformViewsController) {
+        NSObject<FlutterPlatformView> *platformViewProtocolObject = flutterPlatformViewsController->GetPlatformViewByID(child.node.platformViewId);
+        UIView *platformView = [platformViewProtocolObject view];
+        child.accessibilityElements = @[platformView];
```

Comment:
What about `accessibilityContainer` on platfromView? Doesn't that have to link back?

Reply:
I don' think we need to set the accessibilityContainer on the platform view since it is already an UIView. 
So our a11y tree finds this node for platform views, and we set its accessibilityElements as the platform view. Now the platform view will handle the a11y on however it wants, but in the end, because the `child` here is still part of our tree, it will return when the accessibilityElements is done traversing.

Reply:
My understanding is - and this may be incorrect - that in the iOS tree on iOS each node needs to know its parent (the accessibilityContainer) and it's children, if it has any. In this case, the `platformView` doesn't know what its a11y parent is. This may be a problem if the platform view is contained within a Flutter Scroll view: if you put a11y focus on a node within the platform view, iOS cannot walk up the hierarchy to find the scrolling a11y node and dispatch the scroll event there. At least, that how I think that would work....

Reply:
I think you are correct if traverse the accessibility dynamically. (Using the a11y protocol methods etc.) My understanding is that  If we directly set the accessibilityElements property, iOS would handle the traversal and we don't need to worry about the parents at all. I might be wrong, but when I test with google map view inside a list view it works fine (it goes into the google map then goes out). Is there something else you can think of that might fail if parent is not set? I can try them out.

Reply:
So I tried to switch the code to make the accessibility dynamically traverse through the platform view. It indeed won't pick up the parent as you mentioned. And we'd have to access the platform view's code to actually do it. Now the platform view is completely controlled by the author of the individual plugin so I think we should not touch it at all since they might want to do something else within their accessibility. I guess the safer way to do it would just be statically assign the `accessibilityElements`, and let iOS work its magic.